src/components/common/Tabs/Tabs.tsx

```diff
@@ -40,7 +40,7 @@ const Tabs: React.FC<Props> = ({ tabItems }) => {
           />
         ))}
       </ul>
-      <TabContent tabItems={tabItems} activeTab={activeTab} />
+      <TabContent content={tabItems.find((tab) => tab.id === activeTab)?.content} />
```

Comment:
Array positions are guaranteed, so maybe in this case it's probably simpler to save the tabcontent `index` position and not the ID in local state. Then render it here as:
```tsx
const [activeTab, setActiveTab] = useState(0)
const { content } = useMemo((): TabItemType => tabItems[activeTab], [activeTab])
...
<TabContent content={content} />
...
```

please check this first

Reply:
though tbh, i think this entire thing is over engineered - we only have 2 tabs. No need for mapping etc but it's probably fine for now as it's still small. 

Remember that all these looping things etc, while cool, take computing power and when we can avoid them, we should. Obviously some are faster than others but in this case to simple render tab content on 2 tabs it's over kill to use `Array.prototype.find(...)` and `.map` on a likely tiny array. 

Though I say this and I think that the v1 `useTabs` appraoch i want to reuse is probably similar :sweat_smile:

Reply:
here is the full rework:
```tsx
const Tabs: React.FC<Props> = ({ tabItems }) => {
  // default to first tab in array position
  const [activeTab, setActiveTab] = useState(0)
  const { content } = useMemo((): TabItemType => tabItems[activeTab], [activeTab])

  return (
    <Wrapper>
      <ul>
        {tabItems.map(({ id, title, activeColor }, index) => (
          <TabItem
            key={id}
            title={title}
            // set activeTab by it's array position
            onTabClick={(): void => setActiveTab(index)}
            activeColor={activeTab === id && activeColor}
          />
        ))}
      </ul>
      <TabContent content={content} />
    </Wrapper>
  )
}
```

Reply:
@W3stside Refactored it now, to use the index over the passed tabItems array of objects, instead of a specific ID of each Array object.

As for over engineering: I think the goal is to have a general purpose tabs component. In this particular instance these are 2 tabs, but in theory there could be multiple.

Reply:
ok, well then it's fine. but it is just a rehash of [what we already have](https://github.com/gnosis/dex-react/blob/michel-comments/src/hooks/useTabs.tsx) 

Like I mentioned before it's a good working exercise so all good, but let's not reinvent the wheel where we don't need to

Reply:
I understand. Do we want to replace this with the existing logic?

Reply:
nah, let's do that another time. I think you deserve to merge all this as you busted your ass on it. But eventually either look at how we can re-emulate the logic (if we still like it) into yours. 

Nice work on all this btw :+1: